pkg/tsdb/prometheus/prometheus.go

```diff
@@ -162,13 +163,20 @@ func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest)
 		span.SetTag("stop_unixnano", query.End.UnixNano())
 		defer span.Finish()
 
-		value, _, err := client.QueryRange(ctx, query.Expr, timeRange)
+		var results model.Value
 
-		if err != nil {
-			return &result, err
+		switch query.QueryType {
+		case Range:
+			results, _, err = client.QueryRange(ctx, query.Expr, timeRange)
+			if err != nil {
+				return &result, fmt.Errorf("error: %v, for query: %s", err, query.Expr)
```

Comment:
The usual way to wrap error is to have the error at the end "query: %s failed with: %v". This way if there are more wrapping it will be "error1: error2: error3" sort of like a stack trace.

Reply:
Updated in https://github.com/grafana/grafana/pull/39133/commits/c30a552df16e6b29dd6e8b21e100923d8ed096ea. Also thanks for providing context on why is this order better.